<?xml version="1.0"?>
<launch>

  <include file="$(find dynamic_gazebo_models)/launch/empty_world.launch">
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
    <arg name="world_name" value="$(find dynamic_gazebo_models)/worlds/empty.world"/>
  </include>

  <!-- Model Dynamic Manager -->
  <arg name="add_group_srv" value="/model_dynamics_manager/add_control_group"/>

  <node pkg="dynamic_gazebo_models" type="dynamics_manager" name="dynamics_manager" output="screen"/>


  <!-- Flip Door Left -->

  <param name="spawn_door_1" command="$(find xacro)/xacro.py $(find dynamic_gazebo_models)/models/flip_door_right.sdf" />

  <node pkg="gazebo_ros" type="spawn_model" name="gazebo_door1" args="-sdf -param spawn_door_1 -model door_1 -x -3.042358 -y 5.532777 -z 0.3" respawn="false" output="screen">
  </node>

  <!-- Flip Door Right -->

  <param name="spawn_door_2" command="$(find xacro)/xacro.py $(find dynamic_gazebo_models)/models/flip_door_left.sdf" />

  <node pkg="gazebo_ros" type="spawn_model" name="gazebo_door2" args="-sdf -param spawn_door_2 -model door_2 -x -4.452335 -y 5.078486 -z 0.3" respawn="false" output="screen">
  </node>

  <!-- Slide Door Right -->

  <param name="spawn_door_3" command="$(find xacro)/xacro.py $(find dynamic_gazebo_models)/models/slide_right.sdf" />

  <node pkg="gazebo_ros" type="spawn_model" name="gazebo_door3" args="-sdf -param spawn_door_3 -model door_3 -x -5.728697 -y -13.045468 -z 0.3" respawn="false" output="screen">
  </node>

  <!-- Slide Door Left -->

  <param name="spawn_door_4" command="$(find xacro)/xacro.py $(find dynamic_gazebo_models)/models/slide_left.sdf" />

  <node pkg="gazebo_ros" type="spawn_model" name="gazebo_door4" args="-sdf -param spawn_door_4 -model door_4 -x -5.728697 -y -12.232026 -z 0.3" respawn="false" output="screen">
  </node>


</launch>